_functions.sh

```diff
@@ -207,22 +207,28 @@ initialize_product_settings() {
 
   # ${PRODUCT_BRANCH} is computed from ${PRODUCT_VERSION} and is equal to the version up to the latest dot
   # and with x added. ex : 3.5.0-M4-SNAPSHOT => 3.5.x, 1.1.6-SNAPSHOT => 1.1.x
-  env_var PRODUCT_BRANCH `expr "${PRODUCT_VERSION}" : '\([0-9]*\.[0-9]*\).*'`".x"
-  env_var PRODUCT_MAJOR_BRANCH `expr "${PRODUCT_VERSION}" : '\([0-9]*\).*'`".x"
-  configurable_env_var "INSTANCE_ID" ""
+  if [[ -v PRODUCT_VERSION ]]; then
+    env_var PRODUCT_BRANCH `expr "${PRODUCT_VERSION}" : '\([0-9]*\.[0-9]*\).*'`".x"
+    env_var PRODUCT_MAJOR_BRANCH `expr "${PRODUCT_VERSION}" : '\([0-9]*\).*'`".x"
+    configurable_env_var "INSTANCE_ID" ""
 
-  if [ -z "${INSTANCE_ID}" ]; then
-    env_var "INSTANCE_KEY" "${PRODUCT_NAME}-${PRODUCT_VERSION}"
-  else
-    env_var "INSTANCE_KEY" "${PRODUCT_NAME}-${PRODUCT_VERSION}-${INSTANCE_ID}"
+    if [ -z "${INSTANCE_ID}" ]; then
+      env_var "INSTANCE_KEY" "${PRODUCT_NAME}-${PRODUCT_VERSION}"
+    else
+      env_var "INSTANCE_KEY" "${PRODUCT_NAME}-${PRODUCT_VERSION}-${INSTANCE_ID}"
+    fi
   fi
+  # docker image should have INSTANCE_KEY  
+  if [ -n "${DEPLOYMENT_DOCKER_IMAGE}" ]; then
+      env_var "INSTANCE_KEY" "docker-${DEPLOYMENT_DOCKER_IMAGE}-${DOCKER_IMAGE_VERSION}"
```

Comment:
You should not lose the ``INSTANCE_ID`` parameter if we want to be able to deploy several time the same image